frontend/src/components/pages/CourseDashboard.js

```diff
@@ -24,8 +68,46 @@ export function CourseDashboard() {
             .catch((error) => setError(error.message));
     }, []);
 
+    useEffect(() => {
```

Comment:
I didn't catch that earlier, but this useEffect is already used above. Maybe you can find a way to combine both of them into a single one? (Look up **async** inside useEffect and **Promise.all**, maybe you can make something work with those).